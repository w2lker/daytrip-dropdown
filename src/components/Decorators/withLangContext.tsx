import React from "react";

function withLangContext <Props, Lang>(Child: React.ComponentType<Props>, Context: React.Context<Lang>, lang: Lang):React.FC<Props> {
  const ContextProvider: React.FC<Props> = (props) => {
    return (
      <Context.Provider value={lang}>
        <Child {...props}/>
      </Context.Provider>
      )
  };
  ContextProvider.displayName = `WithLangContext(${Child.displayName}, ${Context.displayName})`;
  return ContextProvider;
}

export default withLangContext;
