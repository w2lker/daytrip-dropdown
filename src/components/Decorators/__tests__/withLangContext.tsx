import React, {createContext, ReactElement} from "react";
import withLangContext from "../withLangContext";
import {shallow} from "enzyme";

const setup = () => {
  const SampleComponent = ():ReactElement => {
    return (<></>);
  };
  const sampleLang = {
    text: 'Text passes',
  };

  const SampleContext = createContext(sampleLang);
  const sampleProps = {
    customProp: 'customProp-passes'
  };

  return {SampleComponent, SampleContext ,sampleProps, sampleLang};
};

it('pass original props if they are provided', () => {
  const {SampleComponent, SampleContext, sampleLang, sampleProps} = setup();
  const DecoratedComponent = withLangContext(SampleComponent, SampleContext, sampleLang);
  // @ts-ignore
  const component = shallow(<DecoratedComponent {...sampleProps}/>);
  const props = component.find(SampleComponent).props() as typeof sampleProps;
  expect(props.customProp).toBe(sampleProps.customProp);
});

it('actually creates context', () => {
  const {SampleComponent, SampleContext, sampleLang, sampleProps} = setup();
  const DecoratedComponent = withLangContext(SampleComponent, SampleContext, sampleLang);
  // @ts-ignore
  const component = shallow(<DecoratedComponent {...sampleProps}/>);
  const context = component.find('ContextProvider');
  expect(context.length).toBe(1);
  expect(context.prop('value')).toEqual(sampleLang);
});

