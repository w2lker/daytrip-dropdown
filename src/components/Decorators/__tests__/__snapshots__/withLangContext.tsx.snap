// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
"<WithLangContext(undefined, undefined) customProp=\\"customProp-passes\\">
  <SampleComponent customProp=\\"customProp-passes\\" />
</WithLangContext(undefined, undefined)>"
`;
